streams/src/main/java/org/apache/kafka/streams/processor/internals/assignment/SubscriptionInfo.java

```diff
@@ -82,11 +83,21 @@ public SubscriptionInfo(final int version,
                             final UUID processId,
                             final String userEndPoint,
                             final Map<TaskId, Long> taskOffsetSums) {
+        this(version, latestSupportedVersion, processId, userEndPoint, taskOffsetSums, new AtomicInteger(0));
+    }
+
+    public SubscriptionInfo(final int version,
+                            final int latestSupportedVersion,
+                            final UUID processId,
+                            final String userEndPoint,
+                            final Map<TaskId, Long> taskOffsetSums,
+                            final AtomicInteger shutdownRequested) {
```

Comment:
It seems like we can look at this field two ways:
1. if we're really just flagging "shutdownRequested or not", then a `boolean` is fine.
2. if we're creating a general facility for members to send error codes to the assignor, then we should call this field "errorCode" instead, and an `int` is fine.

Reply:
Cool, so maybe this field should be called something else, like "subscription flag"?

Reply:
sure